cloudinit/util.py

```diff
@@ -68,17 +68,29 @@
                    ['running-in-container'],
                    ['lxc-is-container'])
 
-PROC_CMDLINE = None
+_CACHED_RESPONSES = {}
 
-_LSB_RELEASE = {}
 
+try:
+    from functools import lru_cache
+except ImportError:
+    def lru_cache(f):
```

Comment:
This isn't quite right; `lru_cache` is a decorator _factory_, not a decorator. So this should be taking (no) arguments and _returning_ a decorator.